tests/utils.py

```diff
@@ -1,24 +1,42 @@
 import enum
+from dataclasses import dataclass
 from unittest.mock import patch
+from typing import List
+from pydantic import field_validator
 
 
 class PatchedCalls(enum.Enum):
     GUIDANCE = "guidance._program.Program.__call__"
     # OPENAI_MODEL_LIST = 'openai.models.list'
-    OPENAI_MODEL_LIST = "openai.api_resources.model.Model.list"
-    OPENAI_CHAT_COMPLETION = (
-        "openai.api_resources.chat_completion.ChatCompletion.create"
-    )
-    OPENAI_EMBEDDING_CREATE = "openai.Embedding.create"
+    OPENAI_MODEL_RETRIEVE = "openai.resources.models.Models.retrieve"
+    OPENAI_CHAT_COMPLETION = "openai.resources.chat.completions.Completions.create"
+    OPENAI_EMBEDDING_CREATE = "openai.resources.embeddings.Embeddings.create"
+
+
+@dataclass
+class OpenaiChatCompletionMessageMock(object):
+    content: str
+
+
+@dataclass
+class OpenaiChatCompletionChoiceMock(object):
+    message: OpenaiChatCompletionMessageMock
```

Comment:
Do you have a general strategy for how to mock openai responses in the future? If we only care about chat completions, this seems ok to start with, but there must be libraries that handle this for more than one endpoint in a cleaner way, eg you've mentioned https://github.com/polly3d/mockai. Also compare to what we're doing in pytest for LSE: https://github.com/HumanSignal/label-studio-enterprise/blob/develop/label_studio_enterprise/lse_tests/utils.py#L221

Reply:
It makes sense to have a single testing tool to mock LLM clients. We can use them to `label-studio-sdk` for example, to reuse across different apps